addons/web/static/src/legacy/js/fields/relational_fields.js

```diff
@@ -427,7 +427,7 @@ var FieldMany2One = AbstractField.extend({
         var self = this;
         return {
             res_model: this.field.relation,
-            domain: this.record.getDomain({fieldName: this.name}),
+            domain: this.record.getDomain(this.recordParams),
```

Comment:
Is this also taking into account possible domains in the field? Maybe you need to do a "combination" of both.

Reply:
OK, so the problem is that in this scope `{fieldName: this.name}` doesn't have the same value as `this.recordParams[fieldName]`?

Reply:
The problem is that the object passed has not the viewType, so it is getting the viewType of the element, in this case `"kanban"` the kanban has not a domain defined so it is `"[]"`. By doing this change the viewType is being get from the `"form"` and it has the domain selected. 

As you can see, for the context is used this.recordParams, IDK why here it wasn't used... https://github.com/odoo/odoo/blob/b8a5175b6c92749bd3bb7b9f869b1ecff78e133f/addons/web/static/src/legacy/js/fields/relational_fields.js#L431

Reply:
Thanks for all the explanations. I think they should be added to the commit message / main comment.